### ISSUE TITLE
Node.js: Need to setup cljs for cljs.user ns

### ISSUE THREAD

**Issue by mfikes** [ScalaConsultants/replumb#22]
There is [a line](https://github.com/ScalaConsultants/replumb/blob/master/src/cljs/replumb/repl.cljs#L421) that attempts to set up `cljs.user` which reads:

```clojure
(set! (.. js/window -cljs -user) #js {})
```

I found that, `cljs` itself was not even set up for some reason, with the following error appearing:

```
Initializing REPL environment...
Exception caught in read-eval-call:  #object[TypeError TypeError: Cannot set property 'user' of undefined]
Calling back!
 {:opts {:verbose true},
 :data {},
 :res
 {:error
  #object[TypeError TypeError: Cannot set property 'user' of undefined]}}

{:success? false, :form nil, :error #object[TypeError TypeError: Cannot set property 'user' of undefined]}
```

and that I can work around this by having this in my client code:

```clojure
(set! (.. js/window -cljs) #js {})
```

**Comment by arichiardi** [comment]
Thanks Mike, yes I was expecting a few quirks. I remember having seen [this line](https://github.com/kanaka/cljs-bootstrap/blob/master/src/cljs_bootstrap/core.cljs#L85) and having bookmarked it. Probably @kanaka can explain this better.

My thought is that I can provide a custom `init-fn` for whatever it is needed "client-side". Does it seem reasonable?

**Comment by mfikes** [comment]
@arichiardi Sure, the ability to run environment-specific initialization seems reasonable.

**Comment by kanaka** [comment]
@arichiardi @mfikes it's starting to get a bit fuzzy now, but I believe I need the faction in order to get the starting namespaces to work correctly for three different cases: repl in a browser, repl in a regular js/node build, and repl in a js/node build that has been compiled into a single file. It was a bit touchy to get all three to work correctly because the meaning of "global/window" is different in each case. That's what I remember anyway.

**Comment by arichiardi** [comment]
Yep I gathered that from your code and I am going to add it for sure when *target* will be specified as nodejs...same exact code as in your cljs-bootstrap :smile:

**Comment by arichiardi** [comment]
Closed by #30.
